### PR TITLE
trivial: Add pre-commit hooks for style

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

Example potential output:
```
$ git commit -m "foo"
Check Yaml...........................................(no files to check)Skipped
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check docstring is first.................................................Passed
Check for merge conflicts................................................Passed
Mixed line ending........................................................Passed
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted contrib/generate-version-script.py
All done! ✨ 🍰 ✨
1 file reformatted.

check for null / false return mixmatch...................................Failed
- hook id: check-null-false-returns
- exit code: 1

src/fu-engine.c line 299 got FALSE, which is not valid
```

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    -   id: mixed-line-ending
+        args: [--fix=lf]
+-   repo: https://github.com/ambv/black
+    rev: 20.8b1
+    hooks:
+    - id: black
+-   repo: local
+    hooks:
+    - id: check-null-false-returns
+      name: check for null / false return mixmatch
```
Comment (hughsie) [inline .pre-commit-config.yaml:19]: mismatch?

**Review by hughsie — DISMISSED** [review]
I think this is an excellent idea; I think we should give it a go in master for a few weeks and see how it works in reality. It certainly seems quick enough to not get in my way.

**Comment by hughsie** [comment]
Oh, you should probably add yourself to the MAINTAINERS file too.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (superm1) [reply .pre-commit-config.yaml]: fixed, thanks

**Comment by superm1** [comment]
> I think this is an excellent idea; I think we should give it a go in master for a few weeks and see how it works in reality. It certainly seems quick enough to not get in my way.

Here's real world evidence of it working - while I was gluing it to CI I found I messed up the CI jobs thanks to it :)

```
$ git commit
Don't commit to branch...................................................Passed
Check Yaml...............................................................Failed
- hook id: check-yaml
- exit code: 1

while scanning a simple key
  in ".github/workflows/main.yml", line 19, column 9
could not find expected ':'
  in ".github/workflows/main.yml", line 20, column 9

Check JSON...........................................(no files to check)Skipped
Check for broken symlinks............................(no files to check)Skipped

Check Xml............................................(no files to check)Skipped
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check docstring is first.............................(no files to check)Skipped
Check for merge conflicts................................................Passed
Mixed line ending........................................................Passed
black................................................(no files to check)Skipped
check for use of any deprecated items....................................Passed
check for null / false return mistmatch..................................Passed
check shellscript style..................................................Passed
```

**Review by hughsie — APPROVED** [review]

**Pull request merged**